pydantic/fields.py

```diff
@@ -272,13 +296,13 @@ def infer(
 
         if isinstance(value, FieldInfo):
             field_info = value
-            value = field_info.default
+            value = field_info.default_factory if field_info.default_factory is not None else field_info.default
```

Comment:
This looks a bit weird here, shouldn't we use

```suggestion
            value = field_info.default_factory() if field_info.default_factory is not None else field_info.default
```

or otherwise modify the logic here? I know `value` being a callable won't break the logic below, but it's not what you would expect and could cause errors in future?

Reply:
Of course ! Sorry forgot to change it here